### PR TITLE
Created loadingPage component

### LINKED ISSUE #2: Loading Screen
Create a loading screen to display over the app when going to a new section

Use png for now, eventually this will be switched to svg

Mobile View:
image

Desktop View:
Blow up image and center it

### PR DESCRIPTION
**Pull request by trevortippery**
This PR closes #2

I created the loading page component.

- Loading: This will be used to display over the app when loading into a new section of the app

Here is a screenshot of code usage example in the Home page:

Screenshot 2024-03-02 at 2 33 42 PM

**Loading **Component****

**State Usage:**

- Utilizes the `useState` hook to manage the loading state.
- Initializes `isLoading` state to true to signify the loading process.

**Component Declaration:**

- Defines a functional component named `Loading`.
- Implements a `useEffect` hook to update the loading state after a specified duration. This will be changed in the future but for now I simulated a specified duration of loading time by using 4 seconds or (4000 milliseconds).
- Returns JSX elements representing the loading spinner or icon.

**Rendering:**

- Conditionally renders the loading indicator based on the `isLoading state`. In this instance the loading indicator is the placeholder logo that I converted from a PNG to a SVG for now. I did this so that the scaling of the logo stayed consistent when testing/using different sized devices (i.e. a laptop, large to small mobile device, or a tablet).

The `Loading` component aids in user experience by providing feedback that the application is working on a task and helps alleviate user uncertainty during loading periods.

The two placeholder logos were added to the assets folder which were converted from PNG to SVG for better scaling. There has been some lost of quality by doing the conversion in the process.

**Feedback**
If there are any areas that require adjustments your feedback would be very helpful.

### CODE REVIEW

File: src/components/Loading/Loading.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react'
+import styles from './Loading.module.scss';
+import logo from '../../assets/Temp_Roadwatch_logo.svg';
+// import logo2 from '../../assets/Temp_Roadwatch_logo-2.svg'
+
+export function Loading(): JSX.Element {
+  const [isLoading, setIsLoading] = useState(true);
```
Comment (Linkaloo) [inline src/components/Loading/Loading.tsx:7]: So, I don't think we need this. This seems to be acting like a set loading time to me, correct me if I'm wrong.
But we want the parent component to request something and then if that takes time then it'll render this. 
Essentially the parent has an isLoading flag not the loading component itself. So I think you can just remove all the JS here and just have a div with the logo centered and its good.

File: src/components/Loading/Loading.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react'
+import styles from './Loading.module.scss';
+import logo from '../../assets/Temp_Roadwatch_logo.svg';
+// import logo2 from '../../assets/Temp_Roadwatch_logo-2.svg'
```
Comment (Linkaloo) [inline src/components/Loading/Loading.tsx:4]: Not sure what this is, but remove anything you don't need

File: src/components/Loading/Loading.tsx (hunk 1 above)
Comment (trevortippery) [reply src/components/Loading/Loading.tsx:7]: Yeah, I was simulating a loading time for 4 seconds, but certainly I will just have the loading be a div with a logo centered in the middle.

File: src/components/Loading/Loading.tsx (hunk 2 above)
Comment (trevortippery) [reply src/components/Loading/Loading.tsx]: The excess import statements were removed. I was testing for both logos at the time.

**Review by Linkaloo — APPROVED** [review]

File: src/components/Loading/Loading.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import styles from './Loading.module.scss';
+import logo from '../../assets/Temp_Roadwatch_logo.svg';
+
+export function Loading(): JSX.Element {
```
Comment (KingJordan152) [inline src/components/Loading/Loading.tsx:4]: I recommend exporting this as `default` and then modifying the `index.ts` file accordingly.
```suggestion
export default function Loading(): JSX.Element {
```

File: src/components/Loading/Loading.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  return (
+    <div className={styles['Loading__container']}>
+      <img src={logo} alt="Loading..." className={styles['Loading__logo']} />
```
Comment (KingJordan152) [inline src/components/Loading/Loading.tsx:7]: The `alt` text should be a bit more descriptive. Maybe you could say "Loading indicator" or "Loading indicator displaying the Roadwatch logo."

For more information on this please check out this article: https://wcag.com/blog/good-alt-text-bad-alt-text-making-your-content-perceivable/

File: src/components/index.ts [hunk 5]
```diff
@@ -1 +1,2 @@
 export { default } from './Navbar/Navbar';
+export * from './Loading/Loading';
```
Comment (KingJordan152) [inline src/components/index.ts:2]: If you use a default export in the `Loading` component, your export here should look similar to the Navbar's export. 

For instance, you could do the following: 
```suggestion
export { default as Navbar} from './Navbar/Navbar';
export {default as Loading} from './Loading/Loading';
```

**Review by KingJordan152 — CHANGES_REQUESTED** [review]
Overall, this code looks very clean and works as intended. I especially appreciate your thorough PR message and justifications for certain decisions. Keep up the awesome work!

I made a few requests below pertaining to alt text and exporting conventions. Please take a look at them when you have the chance.

File: src/components/Loading/Loading.tsx (hunk 4 above)
Comment (trevortippery) [reply src/components/Loading/Loading.tsx]: I ended up using "Loading indicator displaying the Roadwatch logo."

**Review by KingJordan152 — APPROVED** [review]
Everything looks good, great work! :sunglasses: 
Thank you for your time.

**Review by Aligary — APPROVED** [review]
Thanks for taking the liberty of adding an svg for the icons! I am going to approve this PR so that we can get this change merged.

**Comment by trevortippery** [comment]
Thank you everyone.

**Pull request merged**